### PR TITLE
Issue 83

### LINKED ISSUE #83: Implement New 1st-Party Login & Sign Up Flow
https://github.com/KeiserCorp/Keiser.Metrics.API/issues/390

### PR DESCRIPTION
**Pull request by ish2028**
Closes #83

### CODE REVIEW

File: src/session.ts [hunk 1]
```diff
@@ -103,6 +129,17 @@ export interface StrengthMachineToken extends JWTToken{
   type: 'machine'
 }
 export module Authentication {
+  export async function url (connection: MetricsConnection) {
```
[the remark this answers is not in the document]
Comment (bayssmekanique) [reply src/session.ts:132]: Where would this be used?

**Review by bayssmekanique — CHANGES_REQUESTED** [review]
Some slight syntax changes needed for clarity, noted in comments.

- [ ] Move `MetricsSSO` and `MetricsAdmin` into their own files so that their dependencies can be tree shaken from the main build.  I'll make the necessary changes to the build process before merging.
- [ ] I don't like moving all the test to use `MetricsSSO` by default. Please revert all tests and lets figure out a way to perform tests against the API that doesn't require essentially running as `root` for all the tests.  I suspect we'll need to setup some helper functions that can perform specific actions as `MetricsSSO` during the setup, but allow the test to be performed via the core `Metrics` export as much as possible.

**Comment by ish2028** [comment]
> * [ ]  I don't like moving all the test to use `MetricsSSO` by default. Please revert all tests and lets figure out a way to perform tests against the API that doesn't require essentially running as `root` for all the tests.  I suspect we'll need to setup some helper functions that can perform specific actions as `MetricsSSO` during the setup, but allow the test to be performed via the core `Metrics` export as much as possible.

I basically have it they way you explained.  I am using `MetricsSSO`, which extends the core `Metrics`.  My functions in `test/persistent/user.ts` contain the functions that talk with `MetricsSSO` functions and then call a core `Metrics` function to be authenticated.  But just so I am clear you want me to have helper functions that I pass `Metrics`, that will create an instance of `MetricsSSO` without exposing `MetricsSSO` to the test?

**Comment by bayssmekanique** [comment]
> > * [ ]   I don't like moving all the test to use `MetricsSSO` by default. Please revert all tests and lets figure out a way to perform tests against the API that doesn't require essentially running as `root` for all the tests.  I suspect we'll need to setup some helper functions that can perform specific actions as `MetricsSSO` during the setup, but allow the test to be performed via the core `Metrics` export as much as possible.
> 
> I basically have it they way you explained. I am using `MetricsSSO`, which extends the core `Metrics`. My functions in `test/persistent/user.ts` contain the functions that talk with `MetricsSSO` functions and then call a core `Metrics` function to be authenticated. But just so I am clear you want me to have helper functions that I pass `Metrics`, that will create an instance of `MetricsSSO` without exposing `MetricsSSO` to the test?

Yes, I want a helper function that will spit out an authenticated `Metrics` instance without having to import `MetricsSSO`.

**Pull request merged**